### DISCUSSION TITLE
Print

### DISCUSSION THREAD

**Discussion by oaslsjcnndq** [Budibase/budibase#2944] [Help] [answered]
Is possible to add print button ? To print like report

**Comment by berma83** [comment]
Hi @oaslsjcnndq,

You can add an embed html print button.

Ex.
```
<!DOCTYPE html>
<html>
<body>
<button onclick="window.print()">Print this page</button>
</body>
</html>
```

https://user-images.githubusercontent.com/53205779/136665716-2e170784-69ba-40d3-b381-207173b41234.mp4

**Comment by mjashanks** [comment]
We would really like to support this sometime soon-ish, but it's not on the roadmap right now.

There's a discussion on it here: https://github.com/Budibase/budibase/discussions/1696

Please upvote and add your thoughts in here :)

**Comment by kevintosh** [comment, marked as answer]
This is maybe very old, but I hacked a solution.

I want a very simple "Print" button that calls the system print dialog.

Since the normal Button element from Budibase does not support a JavaScript action (which would be great if it did) I just added a dummy button to have the button exist in the first place, then hid it with a Condition (if 1=1) and then added an Embed element with exact the same html code (got it from inspector), plus I added an onclick action:

`<button onclick="window.print()" class="spectrum-Button spectrum-Button--sizeM spectrum-Button--primary svelte-1gv5n3y spectrum-Button--quiet c0559966773234991a3d43bc7904da611-dom" contenteditable="false" draggable="false"> Imprimir</button>`

![Screenshot 2023-11-28 at 10 34 50 PM](https://github.com/Budibase/budibase/assets/715643/b1594e3e-15ee-40b9-8a13-098a642211c8)

![Screenshot 2023-11-28 at 10 35 08 PM](https://github.com/Budibase/budibase/assets/715643/662ecbfe-f6a3-4f7a-8a87-ca68aafd79b2)

It could maybe be a nice idea if budibase would support JS actions on the buttons, just to add some pep to the development ;)

**Comment by Hades1280** [comment]
Hi, i tried this but i guess not work anymore. 

Any other solution to print the whole page?

Thanks

**Comment by mikesealey** [comment]
@Hades1280 

Add a button with an on-click action to update state

![image](https://github.com/user-attachments/assets/c5ef8c0f-558f-4572-8b62-d468f4f8abef)

In the value, switch to javascript and use `parent.window.print()`

![image](https://github.com/user-attachments/assets/8c53691a-ef41-4bc2-a074-86be02d4cdf6)


Because the JS here is evaluated for the preview, the code is effectively being run at any change you make, so it's likely that just typing out that line of JS will open a printing dialogue. This will also be the case when you open the bindings drawer, or make any further changes here.

Save the JS here, and publish. The button will now trigger printing the page.

![Screen Recording 2025-01-24 at 1 38 38 PM](https://github.com/user-attachments/assets/04b5ae8a-3a18-4155-88aa-1086a9c0f4d7)

Please note that while this is possible to do, it is not by any means a tested feature of Budibase, and may be unpredictable or produce unexpected results.

**Comment by Hades1280** [comment]
@mikesealey
It works. Simple lovely, i tried figure out how to use a simple javascript to do this, but more simple is impossible. Thanks for your help and I will try to use it carefully.
